File: pyround1/14.最长公共前缀.py
#
# @lc app=leetcode.cn id=14 lang=python
#
# [14] 最长公共前缀
#

# @lc code=start
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ""
        s1 = min(strs)
        s2 = max(strs)
        for i in range(len(s1)):
            if s1[i] != s2[i]:
                return s1[:i]
        return s1


# @lc code=end

File: 14.最长公共前缀.cpp
/*
 * @lc app=leetcode.cn id=14 lang=cpp
 *
 * [14] 最长公共前缀
 */

// @lc code=start
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty())
            return "";
        string res = strs[0];
        for(int i = 1; i < strs.size(); i++){
            if(strs[i].size() < res.size())
                res = strs[i];
        }
        string result = "";
        for(int i = 0; i < strs.size(); i++)
        {
            int n = res.size();
            while(n>0){
                if(res[n-1] != strs[i][n-1])
                {
                    res.pop_back();
                    n = res.size();
                }
                else
                {
                    n--;
                }
            }
        }
        return res;

    }
};
// @lc code=end

